### DISCUSSION TITLE
Spacy 3 fails on Mac M1 - zsh: illegal hardware instruction

### DISCUSSION THREAD

**Discussion by rtlawton** [explosion/spaCy#9397] [Help: Installation] [answered]
## How to reproduce the behaviour
I have just upgraded to spacy 3.1.3, using "pip install -U spacy"
System reported "Successfully installed spacy-3.1.3"
but any command line use of spacy seems to generate "zsh: illegal hardware instruction."
Error occurs on:
"python -m spacy download en_core_web_sm-3.1.0 "or
"python -m spacy info --markdown"

## Your Environment

* Operating System: Mac OS Big Sur 11.6
* Python Version Used: 3.9.6
* spaCy Version Used: 3.1.0
* Environment Information: Conda environment

**Comment by adrianeboyd** [comment, marked as answer]
It sounds like you've gotten package installs that are mixed between native (arm64) and emulation (x86_64) versions, probably due to mixing pip and conda.

It's probably easiest to keep things straight if you install spacy solely through conda: `conda install -c conda-forge spacy`. It might be easiest to start in a new conda venv to be sure you're getting a clean installation of everything.

If you're not already, I'd recommend using the mac arm64 miniforge conda installer: https://github.com/conda-forge/miniforge#miniforge3. There are mac arm64 packages for spacy and all its dependencies through the `conda-forge` channel.

To make things a lot faster on an M1, also install the new `thinc-apple-ops` package. There's only a `pip` package at this point, let's see, I think if you install it carefully with `python -m pip` in your venv it should end up compiling it the right way (you will need Xcode installed to install it):

```bash
python -m pip install thinc-apple-ops
```

**Reply by isobarbaric** [reply]
I tried this but I still keep getting an error when I try to run the following command:

```
python -m spacy download en_core_web_sm
```

I swear I've been trying to get spacy to work for days now...

**Reply by polm** [reply]
Hello. Rather than posting in multiple old threads, could you please open a new discussion and:

1. show the actual error output you're getting
2. show the output of `spacy info`
3. show the output of this command: `python -c 'import platform; print(platform.platform())'`

I'm sorry you're having trouble with this, but with no information except "it doesn't work" it's hard to help you.
